### PR TITLE
Login, SignUp and SignOut MVVM/MVP

### PR DESCRIPTION
**Pull request by bgouws**
### Login, SignUp and SignOut MVVM/MVP

Login, SignUp and SignOut now all conform to MVMM/MVP, here you can just see the property injection. 

### Breakdown

- Added protocol extension to all three classes

- Did property injection in all three classes

- Fixed problem identified by UITest

- Also updated .gitignore

### CODE REVIEW

**Comment by bgouws** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones removed
==============
+ Play Time/Play Time/Views/LoginView.swift  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bgouws/PlayTime/pullRequest?prid=5374923&bid=17620808)

File: Play Time/Play Time/Views/LoginView.swift [hunk 1]
```diff
@@ -85,30 +87,26 @@ class LoginView: UIViewController, UITextFieldDelegate, UIScrollViewDelegate {
...
-                self.myLoginAnalytics.successfulLogin()
-                self.performSegue(withIdentifier: "ToHomeScreen", sender: self)
-            } else {
-                self.myLoginAnalytics.unsuccessfulLogin()
-                self.showFailureAlert()
-            }
-        }
+        let accManagement = AccountManagementViewModel()
+        accManagement.accountManagementView = self
+        accManagement.accountManagementRepo = AccountManagementModel()
```
Comment (JancoDVTDev) [inline Play Time/Play Time/Views/LoginView.swift:96]: Consider using the word "viewModel" in the variable.

File: Play Time/Play Time/Views/LoginView.swift [hunk 2]
```diff
@@ -85,30 +87,26 @@ class LoginView: UIViewController, UITextFieldDelegate, UIScrollViewDelegate {
     @IBAction func btnLoginTapped(_ sender: UIButton) {
         sender.pulsate()
         //variables
+        activityIndicator.startAnimating()
```
Comment (Zamo22) [inline Play Time/Play Time/Views/LoginView.swift:90]: I see that you start or stop animating and hide/show activity indicator in a few places.
Consider creating a function: `hideLoading()` and `showLoading()` which would do the necessary and then you can just call those where needed. Far more readable

File: Play Time/Play Time/Views/SignUpView.swift [hunk 3]
```diff
@@ -32,40 +34,48 @@ class SignUpView: UIViewController, UITextFieldDelegate {
...
-        } else {
-            self.mySignUpAnalytics.unsuccessfulSignUp()
-            self.showFailureAlert()
-            }
-        }
+        let accManagement = AccountManagementViewModel()
+        activityIndicator.isHidden = false
+        accManagement.accountManagementView = self
+        accManagement.accountManagementRepo = AccountManagementModel()
+        accManagement.signUp(email: email!, password: password!)
```
Comment (Zamo22) [inline Play Time/Play Time/Views/SignUpView.swift:41]: Really dangerous here to force unwrap where user has the ability to leave the form empty. Rather unwrap email and password

**Comment by Zamo22** [comment]
Sorry missed this earlier, few comments

File: Play Time/Play Time/Views/LoginView.swift (hunk 2 above)
Comment (bgouws) [reply Play Time/Play Time/Views/LoginView.swift:90]: Ah yes that would be much more efficient. Will make all the suggested changes in my next pull request.

**Pull request merged**